src/monster.cpp

```diff
@@ -2139,6 +2153,37 @@ void monster::die( Creature *nkiller )
     }
 }
 
+bool monster::mod_mech_power( int amt )
+{
+    if( is_hallucination() || !has_flag( MF_RIDEABLE_MECH ) || !battery_item ) {
+        return false;
+    }
+    amt = -amt;
+    battery_item->ammo_consume( amt, pos() );
+    if( battery_item->ammo_remaining() <= 0 ) {
+        return false;
+    }
+    return true;
+}
+
+bool monster::check_mech_powered()
```

Comment:
```suggestion
bool monster::check_mech_powered() const
```
Doesn't modify the object, so it should be const.